astropy/io/ascii/tests/test_qdp.py

```diff
@@ -258,3 +258,26 @@ def test_get_lines_from_qdp(tmp_path):
         assert file_output[i] == line
         assert list_output[i] == line
         assert text_output[i] == line
+
+
+def test_read_qdpfile_with_oneNO_start_new_sector(tmp_path):
+    example_qdp = """
+    READ Terr 1 2
+    9.0150001E+00  1.5000167E-02 -1.5000167E-02 -2.7721735E-03  5.7468953E-02 -5.7468953E-02  8.2039691E-02  7.8058078E-02
+    9.0500002E+00  1.9999990E-02 -1.9999990E-02  1.1883142E-01  7.9723340E-02 -7.9723340E-02  8.3517993E-02  8.5578051E-02
+    NO
+    1.9920001E+01  2.9999205E-02 -2.9999205E-02  2.1270609E-01  8.9663795E-02 -8.9663795E-02  9.9843595E-02  1.4176735E-01
+    1.9975000E+01  2.5000445E-02 -2.5000445E-02  1.1424498E-01  8.8841605E-02 -8.8841605E-02  1.2776274E-01  1.6331280E-01
+    """
```

Comment:
```suggestion
@pytest.mark.parametrize("spacelike", ["", " ", "   ", "\t"])
def test_read_qdpfile_with_oneNO_start_new_sector(tmp_path, spacelike):
    example_qdp = """
READ Terr 1 2
9.0150001E+00  1.5000167E-02 -1.5000167E-02 -2.7721735E-03  5.7468953E-02 -5.7468953E-02  8.2039691E-02  7.8058078E-02
9.0500002E+00  1.9999990E-02 -1.9999990E-02  1.1883142E-01  7.9723340E-02 -7.9723340E-02  8.3517993E-02  8.5578051E-02
{spacelike}NO
1.9920001E+01  2.9999205E-02 -2.9999205E-02  2.1270609E-01  8.9663795E-02 -8.9663795E-02  9.9843595E-02  1.4176735E-01
1.9975000E+01  2.5000445E-02 -2.5000445E-02  1.1424498E-01  8.8841605E-02 -8.8841605E-02  1.2776274E-01  1.6331280E-01
"""
```
I think the current example does not test the functionality that _any_ spacelike sequence ahead of the `NO` is accepted; this should do it (extend as needed if I've forgotten any spacelikes).